### ISSUE TITLE
Property 'color' not being overwritten by variant styling

### ISSUE THREAD

**Issue by NavyAnt24** [tamagui/tamagui#1473]
### Current Behavior

Currently, when creating your own design system component, let's say you do something like this:

```
export const MyButton = styled(Button, {
  backgroundColor: '$blue10',
  color: 'white',

  variants: {
    inverted: {
       true: {
          backgroundColor: 'white',
          color: 'black',
        },
    },
  } as const,
})
```

While I am seeing the `backgroundColor` property being updated, the `color` property from the variant (which I would expect to used, since I understand a variant to be a more specific identifier), is not being used. Instead, the default `color` property is still used. This is problematic, as I would need to pass the color directly on the component in order for the default color to be overridden.

Changing the position of the variant code in the component definition does not fix the problem.

### Expected Behavior

Properties on variants (in my case, specifically the `color` property), should override the default properties of the component. Specifically, using the example above:

`<MyButton inverted />` should have black text inside the button, not white text.

### Tamagui Version

```markdown
1.45.10
```

### Reproduction

```markdown
CodeSandbox is causing me issues, but here is some simple code to try out which does not work properly:


import { Button, styled } from 'tamagui'

export const MyButton = styled(Button, {
  backgroundColor: 'black',
  color: 'white',

  variants: {
    inverted: {
      true: {
        backgroundColor: 'white',
        color: 'black',
        borderColor: 'black',
      },
    },
  } as const,
})

```
```


### System Info

_No response_

**Comment by NavyAnt24** [comment]
This problem is resolvable by following the guide the [How to Build a Button](https://tamagui.dev/docs/guides/how-to-build-a-button) guide, but personally, I think a sane default would be to allow the `color` property to change the color of the text inside the standard Tamagui button component.

**Comment by natew** [comment]
Hm we had this fixed at one point it was very tricky because technically its two components and youre passing through, i'll double check and see if it regressed.

**Comment by natew** [comment]
this should be fixed in `v1.73.1`
